common-tests/src/gateway.rs

```diff
@@ -0,0 +1,63 @@
+use http_types::headers::AUTHORIZATION;
+use hyper::Method;
+use serde::Serialize;
+use wiremock::{
+    matchers::{method, path},
+    Mock, MockServer, Request, ResponseTemplate,
+};
+
+pub async fn mocked_gateway_server() -> MockServer {
+    let mock_server = MockServer::start().await;
+
+    let projects = vec![
+        Project {
+            id: "id1",
+            account_id: "user-1",
+            name: "user-1-project-1",
+            state: "stopped",
+            idle_minutes: 30,
+        },
+        Project {
+            id: "id2",
+            account_id: "user-1",
+            name: "user-1-project-2",
+            state: "ready",
+            idle_minutes: 30,
+        },
+        Project {
+            id: "id3",
+            account_id: "user-2",
+            name: "user-2-project-1",
+            state: "ready",
+            idle_minutes: 30,
+        },
+    ];
+
+    Mock::given(method(Method::GET))
+        .and(path("/projects"))
+        .respond_with(move |req: &Request| {
+            let Some(bearer) = req.headers.get(&AUTHORIZATION) else {
+                return ResponseTemplate::new(401);
+            };
+
+            let user = bearer.last().as_str().split_whitespace().nth(1).unwrap();
```

Comment:
I think doing a full convert to our JWT is too much to just test the clients. Hence, using the bearer value as is